### PR TITLE
feat: add readme files and scripts

### PR DESCRIPTION
**Pull request by noahwaldner**
Add Readme Files

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,37 +60,24 @@ This will add a folder inside `src/components` with the name `CoolCatsAndKittens
...
-Breaking Changes can be declared with a exclamation mark (!) after the type.
+## Releasing Packages
 
-### GitHub Actions
+We utilize [Changesets](https://github.com/changesets/changesets) to manage our releases.
 
--   GitHub automatically deploys Storybook with your new changes to Netlify and adds a Preview link to the PR.
--   SonarCloud checks the code base for its quality.
--   Cypress tests are run to make sure, you're not breaking anything.
+To bump the version of a Package, run `pnpm release` in the root directory and follow the prompts to select your package and version bump.
```
Comment (SamuelAlev) [inline CONTRIBUTING.md:79]:
```suggestion
To bump the version of a package, run `pnpm changeset` in the root directory and follow the prompts to select your package and version bump.
```

Comment (SamuelAlev) [inline CONTRIBUTING.md:79]: Could you add a paragraph about the description? 
The description should be prefixed with "feat:" / "fix:" / ..., be easily readable and understandable

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,37 +60,24 @@ This will add a folder inside `src/components` with the name `CoolCatsAndKittens
...
+A file containg your release notes will be created in the `.changeset` directory.
 
+After your change is merged to the `main` branch, the change will automatically be added to a release PR.
+Once we merge the release PR, the package will be published to NPM.
```
Comment (SamuelAlev) [inline CONTRIBUTING.md:83]:
```suggestion
Once we merge the release PR, the package will be published to npm.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -50,37 +60,24 @@ This will add a folder inside `src/components` with the name `CoolCatsAndKittens
...
-```
-feat: Add CoolCatsAndKittens Component
-```
+-   `pnpm build:components`
+-   `pnpm build:fondue`
+-   `pnpm build:icons`
+-   `pnpm build:charts`
 
-Breaking Changes can be declared with a exclamation mark (!) after the type.
+## Releasing Packages
```
Comment (SamuelAlev) [inline CONTRIBUTING.md:75]:
```suggestion
## Releasing packages
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -50,37 +60,24 @@ This will add a folder inside `src/components` with the name `CoolCatsAndKittens
...
 -   `CoolCatsAndKittens.stories.tsx` is for your Stories inside Storybook 📄
 
-## Opening a PR
-
-Once you're happy with your new component, it's time to open a new PR. Push your branch to GitHub as a start:
-
-```shell
-git push -u origin feat/cool-cats-and-kittens
-```
+## Building Packages
```
Comment (SamuelAlev) [inline CONTRIBUTING.md:63]:
```suggestion
## Building packages
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -5,24 +5,34 @@
...
+This Storybook instance is configured and served from the `/storybook-docs` package which is caused to proxy and compose all other Storybook instances in the monorepo.
+
+In the background, multiple instances of Storybook will start up, one for each component package in the monorepo.
+
+-   Storybook for `@frontify/fondue-components` will be available on Port 6006
+-   Storybook for `@frontify/fondue` will be available on Port 6008
+
+You can also start the Storybook instances for the individual packages by running `pnpm storybook` in the respective package directory.
+
 ## Create a new Component
```
Comment (SamuelAlev) [inline CONTRIBUTING.md:30]:
```suggestion
## Create a new component
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+<a href="https://snyk.io/test/github/Frontify/fondue">
+<img src="https://snyk.io/test/github/Frontify/fondue/badge.svg" alt="Known Vulnerabilities" />
+</a>
+<a href="https://github.com/Frontify/fondue/blob/main/README.md" title="latest">
+<img alt="latest" src="https://img.shields.io/npm/v/@frontify/fondue/latest.svg" />
+</a>
+
+## Using `@frontify/fondue`
+
+# Getting Started
```
Comment (SamuelAlev) [inline README.md:13]:
```suggestion
# Getting started
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+Welcome to Fondue, The Design System for the [Frontify](https://frontify.com) ecosystem.
```
Comment (SamuelAlev) [inline README.md:15]:
```suggestion
Welcome to Fondue, the design system for the [Frontify](https://frontify.com) ecosystem.
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+## Installation
+
+Add the Fondue Design System Package as a dependency to your project. `pnpm i @frontify/fondue`
```
Comment (SamuelAlev) [inline README.md:19]:
````suggestion
Add the Fondue design system package as a dependency to your project. 
```shell
npm i @frontify/fondue
# or
pnpm i @frontify/fondue
# or
yarn add @frontify/fondue
```
````

File: README.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+## Finding Components
```
Comment (SamuelAlev) [inline README.md:21]:
```suggestion
## Finding components
```

File: README.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+You can browse all available components in the Storybook instance for the [Fondue Design System](https://fondue-components.frontify.com).
```
Comment (SamuelAlev) [inline README.md:23]:
```suggestion
You can browse all available components in the Storybook instance of the [Fondue design system](https://fondue-components.frontify.com).
```

File: README.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+This Storybook contains all Components from all subpackages in the Fondue Monorepo. Each Component will have a Badge in the Top Bar indicating it's current lifecycele state.
+
+For the designers, the documentation is available in the [Fondue Design System Documentation](https://weare.frontify.com/document/1266?#/using-fondue) on weare.
```
Comment (SamuelAlev) [inline README.md:26]:
```suggestion
This Storybook contains all components from all subpackages in the Fondue monorepo. Each component will have a badge in the top bar indicating its current lifecycle state.

For the designers, the documentation is available in the [Fondue design system documentation](https://weare.frontify.com/document/1266?#/using-fondue).
```

File: packages/components/package.json [hunk 12]
```diff
@@ -63,19 +64,22 @@
...
         "@vitest/coverage-v8": "^1.4.0",
         "@vitest/ui": "^1.4.0",
+        "chalk": "^5.3.0",
         "eslint": "^8.57.0",
         "eslint-plugin-notice": "^0.9.10",
         "happy-dom": "^14.7.1",
         "postcss": "^8.4.38",
         "postcss-import": "^16.1.0",
         "prettier": "^3.2.5",
+        "promises": "link:node:fs/promises",
```
Comment (SamuelAlev) [inline packages/components/package.json:74]: Not needed

File: packages/components/scripts/createNewComponent.ts [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+        if (template.subdirectory) {
+            directory = `${componentDirectory}/${template.subdirectory}`;
+            let directoryExists: boolean;
+            try {
+                await access(directory);
+                directoryExists = true;
+            } catch {
+                directoryExists = false;
+            }
+            if (!directoryExists) {
```
Comment (SamuelAlev) [inline packages/components/scripts/createNewComponent.ts:50]:
```suggestion
            if (!existsSync(directory)) {
```

File: packages/components/scripts/createNewComponent.ts [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+                await mkdir(`${componentDirectory}/${template.subdirectory}`);
+            }
+        }
+        await writeFile(`${directory}/${componentName}${template.extension}`, template.content);
```
Comment (SamuelAlev) [inline packages/components/scripts/createNewComponent.ts:54]:
```suggestion
               mkdirSync(`${componentDirectory}/${template.subdirectory}`);
            }
        }
       writeFileSync(`${directory}/${componentName}${template.extension}`, template.content);
```

File: packages/components/scripts/createNewComponent.ts [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+
+    if (directoryExists) {
+        console.error(chalk.red(`Component ${componentName} already exists.`));
+        process.exit(1);
+    }
+
+    const generatedTemplates = templates.map((template) => template(componentName));
+    console.log(generatedTemplates);
+
+    await mkdir(componentDirectory);
```
Comment (SamuelAlev) [inline packages/components/scripts/createNewComponent.ts:37]:
```suggestion
   mkdirSync(componentDirectory);
```

File: packages/components/scripts/templates/__tests__/component.ct.tsx [hunk 16]
```diff
@@ -0,0 +1,23 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { toKebabCase } from '../../transforms';
+import { type ComponentFileBuilderResponse } from '../../types';
+
+export const Setup = (componentName: string): ComponentFileBuilderResponse => ({
+    content: `/* (c) Copyright Frontify Ltd., all rights reserved. */
+import { expect, test } from '@playwright/experimental-ct-react';
```
Comment (SamuelAlev) [inline packages/components/scripts/templates/__tests__/component.ct.tsx:8]:
```suggestion
    content: `/* (c) Copyright Frontify Ltd., all rights reserved. */

import { expect, test } from '@playwright/experimental-ct-react';
```
Else eslint complains on the created file

File: packages/components/scripts/templates/styles/componentStyles.tsx [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+export const Setup = (componentName: string): ComponentFileBuilderResponse => ({
+    content: `/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { sv } from '#/utilities/styleUtilities';
+
+export const ${componentName}Styles = sv({
+    base: "tw-block",
+})
+
+`,
```
Comment (SamuelAlev) [inline packages/components/scripts/templates/styles/componentStyles.tsx:14]:
```suggestion
});
`,
```

File: packages/components/tsconfig.json [hunk 18]
```diff
@@ -21,6 +21,6 @@
             "#/*": ["./src/*"]
         }
     },
-    "include": [".storybook", "src"],
+    "include": [".storybook", "src", "scripts"],
```
Comment (SamuelAlev) [inline packages/components/tsconfig.json:24]: Should be added in the `tsconfig.node.json`

File: packages/components/vite.config.ts [hunk 19]
```diff
@@ -31,7 +31,9 @@ export default defineConfig({
         environment: 'happy-dom',
         setupFiles: ['./src/setupTests.ts'],
         css: true,
+        include: ['./src/**/*.test.ts', './src/**/*.test.tsx'],
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:34]: The include default are good enough as you exclude in the coverage

File: packages/components/vite.config.ts [hunk 20]
```diff
@@ -31,7 +31,9 @@ export default defineConfig({
...
         coverage: {
+            exclude: [...configDefaults.exclude, '**.**', 'scripts/templates/**/*'],
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:36]: Why `**.**`? Is it possible to remove it?

File: storybook-docs/stories/1 - Getting Started.mdx [hunk 21]
```diff
@@ -0,0 +1,47 @@
+# Getting Started
```
Comment (SamuelAlev) [inline storybook-docs/stories/1 - Getting Started.mdx:1]: Same as the previous comments as `README.md`

File: storybook-docs/stories/2 - Contributing.mdx
Comment (SamuelAlev) [inline storybook-docs/stories/2 - Contributing.mdx:1]: Same as the previous comments as `README.md`

File: storybook-docs/stories/Contributing.mdx
Comment (SamuelAlev) [inline storybook-docs/stories/Contributing.mdx:1]: Why duplicated?

File: storybook-docs/stories/Getting Started.mdx
Comment (SamuelAlev) [inline storybook-docs/stories/Getting Started.mdx:1]: Same here why duplication?

File: storybook-docs/stories/Getting Started.mdx [hunk 22]
```diff
@@ -1 +1,47 @@
...
+
+### Using Old Components
+
+If you are using the old components, you can continue to import them directly from the `@frontify/fondue` package.
+
+```tsx
+import { FondueButton } from '@frontify/fondue';
+
+const App = () => (
+    <FondueButton>Click Me</FondueButton>
```
Comment (SamuelAlev) [inline storybook-docs/stories/Getting Started.mdx:44]:
```suggestion
import { Button } from '@frontify/fondue';

const App = () => (
    <Button>Click Me</Button>
```

**Review by SamuelAlev — CHANGES_REQUESTED** [review]

File: storybook-docs/stories/Contributing.mdx
Comment (noahwaldner) [reply storybook-docs/stories/Contributing.mdx:1]: One is the one that is shown in Storybook when the consumer looks at it, the other is the one shown in github for devs

File: storybook-docs/stories/Getting Started.mdx
Comment (noahwaldner) [reply storybook-docs/stories/Getting Started.mdx:1]: One is the one that is shown in Storybook when the consumer looks at it, the other is the one shown in github for devs

File: CONTRIBUTING.md (hunk 4 above)
Comment (noahwaldner) [reply CONTRIBUTING.md:63]: changed

File: CONTRIBUTING.md (hunk 5 above)
Comment (noahwaldner) [reply CONTRIBUTING.md:30]: changed

File: README.md (hunk 6 above)
Comment (noahwaldner) [reply README.md:13]: changed

File: README.md (hunk 7 above)
Comment (noahwaldner) [reply README.md:15]: changed

File: README.md (hunk 9 above)
Comment (noahwaldner) [reply README.md:21]: changed

File: README.md (hunk 10 above)
Comment (noahwaldner) [reply README.md:23]: changed

File: README.md (hunk 11 above)
Comment (noahwaldner) [reply README.md:26]: changed

File: package.json [hunk 23]
```diff
@@ -16,7 +16,9 @@
         "storybook": "concurrently 'pnpm storybook:packages' 'pnpm storybook:main'",
         "storybook:main": "wait-on http://localhost:6006 http://localhost:6007 http://localhost:6008 && pnpm --stream  --filter {storybook-docs} storybook",
         "storybook:packages": "pnpm --parallel --filter {packages/*} storybook --no-open",
-        "typecheck": "pnpm --stream --filter {packages/*} typecheck"
+        "typecheck": "pnpm --stream --filter {packages/*} typecheck",
+        "release": "changeset",
```
[the remark this answers is not in the document]
Comment (noahwaldner) [reply package.json:20]: agree, changed

File: packages/components/package.json (hunk 12 above)
Comment (noahwaldner) [reply packages/components/package.json:74]: removed

File: packages/components/scripts/createNewComponent.ts (hunk 13 above)
Comment (noahwaldner) [reply packages/components/scripts/createNewComponent.ts:50]: changed

File: packages/components/scripts/createNewComponent.ts [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+        directoryExists = false;
+    }
+
+    if (directoryExists) {
+        console.error(chalk.red(`Component ${componentName} already exists.`));
+        process.exit(1);
+    }
+
+    const generatedTemplates = templates.map((template) => template(componentName));
+    console.log(generatedTemplates);
```
[the remark this answers is not in the document]
Comment (noahwaldner) [reply packages/components/scripts/createNewComponent.ts:35]: removed

File: packages/components/scripts/createNewComponent.ts (hunk 14 above)
Comment (noahwaldner) [reply packages/components/scripts/createNewComponent.ts:54]: changed

File: packages/components/scripts/createNewComponent.ts (hunk 15 above)
Comment (noahwaldner) [reply packages/components/scripts/createNewComponent.ts:37]: changed

File: packages/components/scripts/createNewComponent.ts [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+
+    let directoryExists: boolean;
+    try {
+        await access(componentDirectory);
+        directoryExists = true;
+    } catch {
+        directoryExists = false;
+    }
+
+    if (directoryExists) {
```
[the remark this answers is not in the document]
Comment (noahwaldner) [reply packages/components/scripts/createNewComponent.ts:29]: changed

File: packages/components/tsconfig.json (hunk 18 above)
Comment (noahwaldner) [reply packages/components/tsconfig.json:24]: switched

File: packages/components/vite.config.ts (hunk 20 above)
Comment (noahwaldner) [reply packages/components/vite.config.ts:36]: that way it doesn't check the files in the root directory for coverage, such as tailwind config, postcss config etc.

Comment (noahwaldner) [reply packages/components/vite.config.ts]: I modified the selector to more specific

File: packages/components/vite.config.ts (hunk 19 above)
Comment (noahwaldner) [reply packages/components/vite.config.ts]: No, this is separate from the coverage reports. The coverage report defines which source files to ignore for coverage. This line defines which test files to ignore and not execute

File: README.md (hunk 8 above)
Comment (noahwaldner) [reply README.md]: chnaged

File: CONTRIBUTING.md (hunk 1 above)
Comment (noahwaldner) [reply CONTRIBUTING.md]: added

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -5,24 +5,34 @@
...
 
+Run `pnpm storybook` to start the storybook development server.
+
+Storybook will be available on [http://localhost:6010](http://localhost:6010).
+This Storybook instance is configured and served from the `/storybook-docs` package which is caused to proxy and compose all other Storybook instances in the monorepo.
+
+In the background, multiple instances of Storybook will start up, one for each component package in the monorepo.
+
+-   Storybook for `@frontify/fondue-components` will be available on Port 6006
+-   Storybook for `@frontify/fondue` will be available on Port 6008
```
[the remark this answers is not in the document]
Comment (noahwaldner) [reply CONTRIBUTING.md]: added

File: storybook-docs/stories/2 - Contributing.mdx
Comment (noahwaldner) [reply storybook-docs/stories/2 - Contributing.mdx:1]: applied changes

File: storybook-docs/stories/1 - Getting Started.mdx (hunk 21 above)
Comment (noahwaldner) [reply storybook-docs/stories/1 - Getting Started.mdx]: applied changes

File: README.md [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+import { FondueButton } from "@frontify/fondue";
+
+const App = () => <FondueButton>Click Me</FondueButton>;
+```
+
+## Contributing
+
+See [Contribution guidelines](CONTRIBUTING.md) for contributing and local development help.
+
+## Important Links
```
Comment (SamuelAlev) [inline README.md:70]:
```suggestion
## Important links
```

File: packages/components/scripts/templates/__tests__/component.ct.tsx [hunk 28]
```diff
@@ -0,0 +1,24 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { toKebabCase } from '../../transforms';
+import { type ComponentFileBuilderResponse } from '../../types';
+
+export const Setup = (componentName: string): ComponentFileBuilderResponse => ({
+    content: `/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { expect, test } from '@playwright/experimental-ct-react';
+import { ${componentName} } from "../${componentName}";
```
Comment (SamuelAlev) [inline packages/components/scripts/templates/__tests__/component.ct.tsx:10]:
```suggestion
import { expect, test } from '@playwright/experimental-ct-react';

import { ${componentName} } from "../${componentName}";
```
Did you sync the template with the lint fix?

File: packages/components/scripts/templates/component.stories.ts [hunk 29]
```diff
@@ -0,0 +1,34 @@
...
+
+import { type ComponentFileBuilderResponse } from '../types';
+
+// We aren't really in a story
+// eslint-disable-next-line storybook/default-exports
+export const Setup = (componentName: string): ComponentFileBuilderResponse => ({
+    content: `/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Meta, StoryObj } from "@storybook/react";
+import { ${componentName} } from "./${componentName}";
```
Comment (SamuelAlev) [inline packages/components/scripts/templates/component.stories.ts:11]:
```suggestion
import { type Meta, type StoryObj } from "@storybook/react";

import { ${componentName} } from "./${componentName}";
```

File: packages/components/scripts/templates/component.ts [hunk 30]
```diff
@@ -0,0 +1,24 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { toKebabCase } from '../transforms';
+import { type ComponentFileBuilderResponse } from '../types';
+
+export const Setup = (componentName: string): ComponentFileBuilderResponse => ({
+    content: `/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { ReactNode } from "react";
+import { ${componentName}Styles } from './styles/${componentName}Styles';
```
Comment (SamuelAlev) [inline packages/components/scripts/templates/component.ts:10]:
```suggestion
import { type ReactNode } from "react";

import { ${componentName}Styles } from './styles/${componentName}Styles';
```

File: storybook-docs/stories/1 - Getting Started.mdx [hunk 31]
```diff
@@ -0,0 +1,63 @@
+
+# Getting started
```
Comment (SamuelAlev) [inline storybook-docs/stories/1 - Getting Started.mdx:2]:
```suggestion
# Getting started
```

File: storybook-docs/stories/1 - Getting Started.mdx [hunk 32]
```diff
@@ -0,0 +1,63 @@
...
+import { FondueButton } from "@frontify/fondue";
+
+const App = () => <FondueButton>Click Me</FondueButton>;
+```
+
+## Contributing
+
+See [Contribution guidelines](CONTRIBUTING.md) for contributing and local development help.
+
+## Important Links
```
Comment (SamuelAlev) [inline storybook-docs/stories/1 - Getting Started.mdx:59]:
```suggestion
## Important links
```

File: storybook-docs/stories/1 - Getting Started.mdx [hunk 33]
```diff
@@ -0,0 +1,63 @@
...
+
+-   [Storybook](https://fondue-components.frontify.com) – Storybook for previweing fondue components
```
Comment (SamuelAlev) [inline storybook-docs/stories/1 - Getting Started.mdx:61]:
```suggestion
-   [Storybook](https://fondue-components.frontify.com) – Storybook for previewing Fondue components
```

File: README.md [hunk 34]
```diff
@@ -0,0 +1,74 @@
...
+
+-   [Storybook](https://fondue-components.frontify.com) – Storybook for previweing fondue components
```
Comment (SamuelAlev) [inline README.md:72]:
```suggestion
-   [Storybook](https://fondue-components.frontify.com) – Storybook for previewing Fondue components
```

File: packages/components/scripts/templates/__tests__/component.ct.tsx (hunk 28 above)
Comment (noahwaldner) [reply packages/components/scripts/templates/__tests__/component.ct.tsx:10]: not yet

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**